CentralUnit/Models/Lap.cs

```diff
@@ -27,23 +33,45 @@ public long LapTime
 
         public Lap(FinishedEvent finish)
         {
-            end = finish;
+            End = finish;
+        }
+
+        public Lap(UnitDNFEvent dnf)
+        {
+            End = dnf;
+        }
+
+        public Lap(ManualDNFEvent dnf)
+        {
+            End = dnf;
         }
 
-        public Lap(DNFEvent dnf)
+        public void AddPenalties(List<PenaltyEvent> penalties)
         {
-            end = dnf;
+            this.penalties.AddRange(penalties);
+        }
+
+        public void SetDsq(DSQEvent dsq)
+        {
+            if(this.Disqualified)
+            {
+                throw new InvalidOperationException("Cannot set the DSQ event of a lap that already has one");
+            }
+            else
+            {
+                Dsq = dsq;
+            }
         }
 
         public int CompareTo(Lap other)
```

Comment:
Should potential penalties be included in the comparison? A lap that was faster by missing a jink line was actually slower when including penalties, which matters when sorting or comparing for the score board or a local score board.

Reply:
Good point. I think it makes sense to provide two distinct comparisons here